internal/venice-common/src/main/java/com/linkedin/venice/blob/ServerBlobFinder.java

```diff
@@ -26,29 +23,21 @@ public BlobPeersDiscoveryResponse discoverBlobPeers(String storeName, int versio
     BlobPeersDiscoveryResponse response = new BlobPeersDiscoveryResponse();
     try {
       String currentVersionResource = Version.composeKafkaTopic(storeName, version);
-      // Get the partition assignments for the current version
-      List<String> hostNames = new ArrayList<>();
-      for (Partition partition: customizedViewRepository.getPartitionAssignments(currentVersionResource)
-          .getAllPartitions()) {
-        if (partition.getId() == partitionId) {
-          for (Instance instance: partition.getReadyToServeInstances()) {
-            String host = instance.getHost();
-            hostNames.add(host);
-          }
-          break;
-        }
+      // Get the partition assignments for the specific partition and retrieve the host names
+      Partition partition =
+          customizedViewRepository.getPartitionAssignments(currentVersionResource).getPartition(partitionId);
+      if (partition != null) {
+        response.addPartition(partition);
       }
-      response.setDiscoveryResult(hostNames);
     } catch (VeniceException e) {
       response.setError(true);
-      String errorMsg =
-          "Error finding blob for store: " + storeName + ", version: " + version + ", partitionId: " + partitionId;
-      response.setMessage(errorMsg + ".\n Error: " + e.getMessage());
-      LOGGER.warn(errorMsg, e);
+      String errorMsg = String
+          .format("Error finding blob for store: %s, version: %d, partitionId: %d", storeName, version, partitionId);
```

Comment:
Thank you for changing it to `String.format`!

another comment: I think it should be `Error finding peers for blob transfer in store:...` instead of `Error finding blob for store:...`
 (sorry, didn't catch before)